Add plots to QC report

jashapiro opened the pull request:
Here I am mostly adding the Total vs. gene count plot to the QC report. I also fixed the knee plot and made the session info hidden by default, so closing all three of those issues here.

I was also playing around with #54, which I ended up abandoning, but in that process I added a bit of formatting to the tables, including making them not full-width, and striping the rows.

A few things to focus on that I was thinking about as I did them: 
- Please also look at all axis labels and legend titles! (Should I make the knee plot use non-scientific notation?)

- For consistency among reports, I made the mitochondrial percentage go from 0-100% for all reports. This may make lower numbers harder to see in mostly-good plots (where no cells are above 50%, say). I thought the consistency was more important, but we could tweak the color scale if we think it matters.

- The other potentially controversial aesthetic decision I made was to move the figure legends into the figure panels, at least for the knee & UMI/gene plots. In both of these plots, the location of the data is predictable, so in the vast majority of cases the positions I chose will not overlap data. Still, there is a chance they will, so if there is worry, I can move them back out.  (when we add the miQC plot, the top right ought to work as a clear space)

I am attaching an [example report](https://github.com/AlexsLemonade/scpcaTools/files/7262155/SCPCL000001_qc.html.zip) (zipped, b/c github), and the two main figures I worked on below, for reference:
![image](https://user-images.githubusercontent.com/20483/135521936-6079587d-d297-4891-b234-e4a9923f4b8b.png)
![image](https://user-images.githubusercontent.com/20483/135521948-3a975d48-1895-4722-8f86-1019ab1453a6.png)


closes #39
closes #40
closes #19
closes #55

jashapiro commented:
Adding in the `miQC` plot as well, which looks like this (not included in the example report above):

![image](https://user-images.githubusercontent.com/20483/135685703-a0157349-a0dc-483f-a703-5e4096e623c0.png)

Same story here where I moved the legend inside the plot. In most cases the upper right should have no genes, so it ought to be safe, but I can move back outside if that is preferred.

allyhawkins reviewed (commented):
Overall this is coming together really nicely! I just wanted to comment on a few minor things: 

1. I like the placement of the legends and am not overly concerned about if they overlap data. I think for the majority of data where you've placed them should be outside of where the data lie. 
2. We should be consistent with the type of case we choose to use for the axis titles. I think we could use sentence case as we do most everywhere else, and I think you are mostly using that now so I suggested a few changes to make that consistent throughout. 
3. You mentioned in the comment that you made the legend go from 0-100% for mito content. I think we should do the same thing with plotting the miQC plot. Since we always anticipate that axis to be on the same scale.

jashapiro commented:
I updated legends and titles as suggested, using sentence case everywhere, and unifying labels.

I moved what I had put as figure captions to alt text because it was laregely redundant, as noted, but I thought we should have some kind of figure label beyond the title. 

For the knee plot, I chose a more contrasting color scheme, but I didn't want to go to red, because that would imply (to me) that those points failed some QC. So I went with a dark green, and lightened up the grey. I also went with smaller points for the filtered cells. Let me know what you think. 
![image](https://user-images.githubusercontent.com/20483/135902569-8f5d3f92-ba8c-4fc8-9dda-c6d61a74fb3c.png)

Here is a full report file...
[SCPCL000001_qc.html.zip](https://github.com/AlexsLemonade/scpcaTools/files/7280358/SCPCL000001_qc.html.zip)

allyhawkins commented on inst/rmd/qc_report.rmd:
```diff
@@ -103,32 +123,85 @@ if(has_filtered){
+  filtered_sce$prob_compromised <- NULL
+  miQC::plotModel(filtered_sce, model = metadata(filtered_sce)$miQC_model) + 
+    coord_cartesian(ylim = c(0,100)) +
+    labs(x = "Number of genes expressed",
```
```suggestion
    labs(x = "Number of genes detected",
```
Just noticing that we have detected in the other plot so either we should change it here or change it to expressed in the other plot, I'm fine with either.

allyhawkins reviewed (approved):
I like the changes and think the green in the knee plot looks a lot better! I don't necessarily think we need the change in size and have previously been taught not to double encode variables where we have both size and color showing if a cell is considered `passed` or `excluded` but I'll leave that final call up to you. I do think the changes highlight the passed cells which is what we want so I can see an argument for either one.

I had one other minor axis wording change to keep the labels consistent between the `Cell read metrics` and `miQC` plot, but other than that I think this looks good!

jashapiro commented:
> I don't necessarily think we need the change in size and have previously been taught not to double encode variables where we have both size and color showing if a cell is considered `passed` or `excluded` but I'll leave that final call up to you.

Yeah, I don't usually do this, but I thought it was worth it in this case, as I think it solves a problem where the two kinds of points overlap.

allyhawkins commented:
> Yeah, I don't usually do this, but I thought it was worth it in this case, as I think it solves a problem where the two kinds of points overlap.

I would agree, it does help show the separation a lot more. I think for the purposes of this plot its okay to keep it in.
